tests/framework/grid/DataColumnTest.php

```diff
@@ -144,5 +144,53 @@ public function testFilterInput_Array()
             , $result);
     }
 
+    /**
+     * @see DataColumn::$filter
+     * @see DataColumn::renderFilterCellContent()
+     */
+    public function testFilterInput_FormatBoolean()
+    {
+        $this->mockApplication([
+            'components' => [
+                'db' => [
+                    'class' => '\yii\db\Connection',
+                    'dsn' => 'sqlite::memory:',
+                ]
+            ]
+        ]);
+        $columns = [
+            'id' => 'pk',
+            'customer_id' => 'integer',
+        ];
+        ActiveRecord::$db = Yii::$app->getDb();
+        Yii::$app->getDb()->createCommand()->createTable(Order::tableName(), $columns)->execute();
+
+        $grid = new GridView([
+            'dataProvider' => new ArrayDataProvider([
+                'allModels' => [],
+                'totalCount' => 0,
+            ]),
+            'columns' => [
+                0 => [
+                    'attribute' => 'customer_id',
+                    'format' => 'boolean' // does not make sense for this column but should still output proper dropdown list
+                ]
```

Comment:
Even these are just tests, it's good to adhere code style about comma after last array element here too (it's even inconsistent with `dataProvider` section written above).